components/calendar/src/helpers.rs

```diff
@@ -236,6 +236,28 @@ where
     }
 }
 
+pub(crate) fn next_u8<F>(mut index: u8, condition: F) -> u8
+where
+    F: Fn(u8) -> bool,
+{
+    loop {
+        if condition(index) {
+            return index;
+        }
+        index += 1;
+    }
+}
+
+pub(crate) fn final_func<F>(mut index: i32, condition: F) -> i32
```

Comment:
suggestion: rename (the `_func` suffix sounds redundant)
```suggestion
pub(crate) fn final<F>(mut index: i32, condition: F) -> i32
```

Reply:
Final is a reserved word. Added the comment explaining it.